api/v1/routes/product.py

```diff
@@ -113,13 +122,80 @@ async def update_product(
             "price": 29.99,
             "description": "Updated description",
         }
-    """  
+    """
 
     updated_product = product_service.update(db, id=str(id), schema=product_update)
 
     # Prepare the response
     return success_response(
-        status_code = 200,
-        message = "Product updated successfully",
-        data = jsonable_encoder(updated_product)
-    )
\ No newline at end of file
+        status_code=200,
+        message="Product updated successfully",
+        data=jsonable_encoder(updated_product),
+    )
+
+
+@product.get("/{id}")
+async def get_product_detail(
+    id: str,
+    db: Session = Depends(get_db),
+    _: User = Depends(user_service.get_current_user),
+):
+    """
+    Retrieve product detail
+
+    This endpoint retrieve details about a product
+
+    Args:
+        id (UUID): The unique identifier of the product to be retrieved.
+        db (Session): The database session, provided by the `get_db` dependency.
+        _ (User): The currently authenticated user, obtained from the `get_current_user` dependency.
+
+    Returns:
+        ProductDetail: The detail of the product matching the given id
+
+    Raises:
+        HTTPException: If the product with the specified `id` does not exist, a 404 error is raised.
+    """
+
+    product = product_service.fetch(db, id)
```

Comment:
From this line downwards, isn't there a way to make it cleaner. You can find out online because this makes it hard in case of changes that may occur

Reply:
I have checked and checked but I'll still check again

Reply:
Unless I'm allowed to use the normal fastapi response instead of success_response()